libs/ml/src/layers/convolution_2d.cpp

```diff
@@ -54,16 +55,23 @@ Convolution2D<TensorType>::Convolution2D(SizeType const output_channels,
   output = fetch::ml::details::AddActivationNode<TensorType>(activation_type, this,
                                                              name + "_Activation", output);
 
-  // A temporary fix to prevent Conv2d shape computing crash. Dummy width (32) and height (32).
-  // TODO(VH): Split initialisation as in FullyConnected to prevent dummies.
   this->GetNode(weights)->SetBatchOutputShape(
       {output_channels_, input_channels_, kernel_size_, kernel_size_, 1});
-  this->GetNode(input)->SetBatchOutputShape({output_channels_, 32, 32, 1});
+
+  // TODO(ML-470): Preliminary batch shape of a Conv2d layer (channels x 32(h) x 32(w) x
+  // 1(batch) ), is used here now, however, real width and height are to be set later on graph
+  // compilation (when expected Input shape of the Model/Graph is already known). Thus the
+  // convolutional weight init can be done on constructions, but this->input shape has to be
+  // inited only in this->CompleteInitialisation() override.
+  static constexpr SizeType DEFAULT_HEIGHT = 32;
```

Comment:
if these are just placeholders until the real values are set later, why not just set them both to 1?

Reply:
or better yet 0? that way they can't possibly be correct, and if, when debugging, we find that height and width are 0 we will know that they were never overwritten correctly.

Reply:
it could be neither 0 nor 1 otherwise further calculations of output shape are being affected - a convolution of an image with kernel 3*3 and "valid" padding would produce a smaller image, and also it (input image) could not be less then a kernel size. So I have chosen some meaningful image size for this dummy. Anyway it would be corrected during implementation of a charge estimation for convolution (because real input inmage shape affects Conv performance greatly)